Blizzard.eai

```diff
@@ -1,1760 +1,2052 @@
+  set t = null
+endfunction
+
+function GameStartDefault takes nothing returns nothing
+  local timer t = CreateTimer()
+  if game_mode != "ai_only" then
+    call InitCommanderPart()
+    if game_mode == "no_human" then
+      call ForForce(GetPlayersAll(), function RemoveHumanUnitsAndRes)
+      call SendToAllAIs(53,0)
+    endif
+  endif
+  set last_timer = t
+  //Timer that calls MapInit
+  call TimerStart(t,3.0, false, function MapInit)
+  set t = null
+endfunction
+
+//-------------------------------------------------------------------------
+function GameStartDlg takes nothing returns nothing
+  local integer host = GetFirstPlayableHuman()
+  local integer p = host
+  local timer t = null
+  loop
+    if p == -1 or IsPlayerObserver(Player(p)) then
+      set t = CreateTimer()
+      set last_timer = t
+      //Timer that calls MapInit
+      call TimerStart(t,3.0, false, function MapInit)
+      set t = null
+      return
+    endif
+    exitwhen CheckAllyToControl(Player(p))
+    set p = GetNextPlayableHuman(p)
+  endloop
+  if game_mode != "" then
+    call GameStartDefault()
+    return
+  endif
+  set game_start_dialog = DialogCreate()
+  call DialogSetMessage(game_start_dialog, dlghdr_game_type)
+  set commander_mode = DialogAddButton(game_start_dialog, dlgbutton_commander, 67)
+  set no_human_mode = DialogAddButton(game_start_dialog, dlgbutton_no_human, 79)
+  set ai_only_mode = DialogAddButton(game_start_dialog, dlgbutton_ai_only, 78)
+
+  set game_start_trigger = CreateTrigger()
+  call TriggerRegisterDialogEvent(game_start_trigger, game_start_dialog)
+  call TriggerAddAction(game_start_trigger, function GameStartDlgResponse)
+
+  call DialogDisplay(Player(host), game_start_dialog, true)
+endfunction
+
+function LanguageDlgResponse takes nothing returns nothing
+  local button b = GetClickedButton()
+  if language != "English" and language != "" and b != null then
+    set language = "English"
+    call SendToAllAIs(71,0)  //Second language change , some variable assigned by previous language, so back English cannot take effect , it can fix
+    call initLanguageEnglish()
+  endif
+  if false then
+#INCLUDETABLE <Languages.txt> #EFR
+  elseif b == %1_button then
+    set language = "%1"
+    call SendToAllAIs(71,#EVAL{%row-1})
+    call initLanguage%1()
+#ENDINCLUDE
+  endif
+  call PauseAllUnitsBJ(false)
+  call GameStartDlg()
+  set b = null
+endfunction
+
+function LanguageDefault takes nothing returns nothing
+  if false then
+#INCLUDETABLE <Languages.txt> #EFR
+  elseif language == "%1" then
+    call SendToAllAIs(71,#EVAL{%row-1})
+    call initLanguage%1()
+#ENDINCLUDE
+  endif
+  call PauseAllUnitsBJ(false)
+  if game_mode != "ai_only" then
+    call GameStartDlg()
+  endif
+endfunction
+
+function LanguageDlg takes nothing returns nothing
+  local integer host = GetFirstHuman()
+  local integer i = 0
+  call DestroyTimer(last_timer)
+  if debugging and host != -1 and IsPlayerObserver(Player(host)) then

```

Comment:
Moved this debugging to be controlled by the common.eai debugging mode option in master as blizzard.eai option isn't utilized in the best way.